### ISSUE TITLE
Missing requirement: org.testng 6.9.12 requires 'package com.google.common.primitives [16.0.0,17.0.0)' but it could not be found

### ISSUE THREAD

**Issue by missedone** [testng-team/testng-p2#6]
for quick verify the new testng p2 repo, I make the tycho of testng eclipse plugin point to this local p2 repo, and make the plugin depends on new TestNG OSGi bundle. then I got following error when build:

```
[ERROR] Cannot resolve project dependencies:
[ERROR]   Software being installed: org.testng.eclipse 6.9.13.qualifier
[ERROR]   Missing requirement: org.testng 6.9.12 requires 'package com.google.common.primitives [16.0.0,17.0.0)' but it could not be found
[ERROR]   Cannot satisfy dependency: org.testng.eclipse 6.9.13.qualifier depends on: bundle org.testng 0.0.0
```

steps to reproduce this issue:
1. add local testng p2 repo for tycho in parent pom.xml
  ```xml
        <repository>
                <id>testng-p2-repo</id>
                <layout>p2</layout>
                <url>file:///Users/nick/workspace/myproject/testng-p2/build/updatesite</url>
         </repository>
  ```
2. make TestNG eclipse plugin depends on TestNG OSGi bundle in `testng-eclipse-plugin/META-INF/MANIFEST.MF`
  
  ```
Require-Bundle: org.eclipse.ui,
 org.eclipse.core.runtime,
 org.eclipse.core.expressions,
 org.eclipse.core.resources,
 org.eclipse.compare,
 org.eclipse.debug.core,
 org.eclipse.debug.ui,
 org.eclipse.jface,
 org.eclipse.jface.text,
 org.eclipse.jdt.core,
 org.eclipse.jdt.debug.ui,
 org.eclipse.jdt.junit,
 org.eclipse.jdt.launching,
 org.eclipse.jdt.ui,
 org.eclipse.ui.ide,
 org.eclipse.ui.workbench.texteditor,
 org.eclipse.search,
 org.eclipse.ltk.core.refactoring,
 org.eclipse.ltk.ui.refactoring,
 org.eclipse.core.variables,
 org.junit,
 org.eclipse.ui.editors,
 org.testng,
 com.beust.jcommander
  ```

**Comment by missedone** [comment]
'package com.google.common.primitives' of guava is the transitive dependency of the the optional dependency guice

```
compile - Dependencies for source set 'main'.
+--- org.beanshell:bsh:2.0b4
+--- com.beust:jcommander:1.48
+--- org.apache.ant:ant:1.7.0
|    \--- org.apache.ant:ant-launcher:1.7.0
+--- junit:junit:4.12
|    \--- org.hamcrest:hamcrest-core:1.3
+--- org.yaml:snakeyaml:1.15
\--- com.google.inject:guice:4.0
     +--- javax.inject:javax.inject:1
     +--- aopalliance:aopalliance:1.0
     \--- com.google.guava:guava:16.0.1
```

**Comment by missedone** [comment]
we need to figure out a way, either modify the TestNG OSGi head to remove this optional import package, or make tycho to ignore this optional import package if can't resolve.

**Comment by missedone** [comment]
ping @mschreiber

**Comment by juherr** [comment]
I don't know if it will help but the dependency to Guava was an issue and it is fixed now.

**Comment by missedone** [comment]
hmm, it's supposed to be fixed in 6.9.13, the OSGi head is correct in build.gradle, but since the 6.9.13.x were released by kobalt (not gradle) which used an out-of-date MANIFEST.MF, https://github.com/cbeust/testng/issues/1190#issuecomment-252390348

so, I'm thinking if we need to correct that (means overwrite) the OSGi head for old testng releases (e.g. 6.9.12, 6.9.13, etc.)

**Comment by missedone** [comment]
closed by #9
